server/app/auth/CiviFormProfile.java

```diff
@@ -30,27 +31,46 @@ public class CiviFormProfile {
   private final HttpExecutionContext httpContext;
   private final CiviFormProfileData profileData;
   private final SettingsManifest settingsManifest;
+  private final AccountRepository accountRepository;
 
   public CiviFormProfile(
       DatabaseExecutionContext dbContext,
       HttpExecutionContext httpContext,
       CiviFormProfileData profileData,
-      SettingsManifest settingsManifest) {
+      SettingsManifest settingsManifest,
+      AccountRepository accountRepository) {
     this.dbContext = Preconditions.checkNotNull(dbContext);
     this.httpContext = Preconditions.checkNotNull(httpContext);
     this.profileData = Preconditions.checkNotNull(profileData);
     this.settingsManifest = Preconditions.checkNotNull(settingsManifest);
+    this.accountRepository = Preconditions.checkNotNull(accountRepository);
   }
 
   /** Get the latest {@link ApplicantModel} associated with the profile. */
   public CompletableFuture<ApplicantModel> getApplicant() {
+    if (settingsManifest.getNewApplicantUrlSchemaEnabled()
+        && profileData.containsAttribute(ProfileFactory.APPLICANT_ID_ATTRIBUTE_NAME)) {
+      long applicantId =
+          profileData.getAttribute(ProfileFactory.APPLICANT_ID_ATTRIBUTE_NAME, Long.class);
+      return accountRepository
+          .lookupApplicant(applicantId)
+          .thenApply(
+              optionalApplicant -> {
+                return optionalApplicant.orElseThrow();
+              })
```

Comment:
**nitpick:** 
```suggestion
          .thenApply(Optional::orElseThrow)
```

Reply:
Thanks for this suggestion. But based on your other suggestion (throwing meaningful exceptions in `.orElseThrow()`), I am retaining this syntax.